src/components/_button.scss

```diff
@@ -180,6 +213,21 @@
   font-size: $fz-l;
 }
 
+@mixin sui-button-icon--small {
```

Comment:
In this escenario I think is a better approach to include one mixin or the other rather than re-declare the same sass var depending of the value of another var:
```
@if $size == 'small' {
     @include sui-icon--small;
}
```